src/elections/crud.py

```diff
@@ -0,0 +1,68 @@
+import dataclasses
+import logging
+from datetime import datetime
+
+import database
+import sqlalchemy
+from elections.tables import Election
+from officers.constants import OfficerPosition
+from officers.tables import OfficerInfo, OfficerTerm
+
+_logger = logging.getLogger(__name__)
+
+async def get_election(db_session: database.DBSession, election_slug: str) -> Election | None:
+    query = sqlalchemy.select(Election)
+    query = query.where(Election.slug == election_slug)
+    result = (await db_session.execute(query)).scalar()
+    db_session.commit()
```

Comment:
Here's an interesting question: how do we know if a crud function has committed current db session or not? The unfortunate answer is that you have to read the function every time. 

For this reason, it's best to never commit in a crud function, so that the functions can be used in different contexts. Instead, transactions should be managed on a per-endpoint basis.